workflows/rnasplice.nf

```diff
@@ -61,23 +123,494 @@ def multiqc_report = []
 
 workflow RNASPLICE {
 
+    // Create channel for software versions (will be added to throughout pipeline)
     ch_versions = Channel.empty()
 
     //
-    // SUBWORKFLOW: Read in samplesheet, validate and stage input files
+    // SUBWORKFLOW: Uncompress and prepare reference genome files
     //
-    INPUT_CHECK (
-        ch_input
+    PREPARE_GENOME (
+        params.fasta,
+        params.gtf,
+        params.gff,
+        params.transcript_fasta,
+        params.star_index,
+        params.salmon_index,
+        params.gff_dexseq,
+        params.suppa_tpm,
+        is_aws_igenome,
+        prepare_tool_indices,
+        params.source,
+        params.gencode
     )
-    ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
+    ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)
 
     //
-    // MODULE: Run FastQC
+    // SUBWORKFLOW: Read in samplesheet, validate and stage input files
+    //
+    switch (params.source) {
```

Comment:
This can be condensed a bit I think using channel manipulation

Reply:
Thanks for the tip, agreed it looks a bit messy, will look into refactoring in next release.